src/main/kotlin/org/rust/ide/inspections/lints/RsUnusedImportInspection.kt

```diff
@@ -189,7 +189,8 @@ private fun isItemUsedInOtherMods(item: RsNamedElement, importName: String, useS
             RsVisibility.Restricted(importMod)
         }
         is RsVisibility.Restricted -> visibility
-        RsVisibility.Public -> return true
+        // we handle public imports in binary crates only, so this is effectively `pub(crate)` import
+        RsVisibility.Public -> null
```

Comment:
Maybe we could simplify things around it by returning a honest `RsVisibility.Restricted(crateRoot)`